slick/src/main/scala/slick/util/InternalArrayQueue.scala

```diff
@@ -0,0 +1,175 @@
+package slick.util
+
+import java.util
+
+private[util] final class InternalArrayQueue[E >: Null <: AnyRef](capacity: Int) {
```

Comment:
Was this ported from some Java library? If so, it needs attribution (and a compatible license in the first place).